share/pvss/pvss.go

```diff
@@ -0,0 +1,172 @@
+// Package pvss implements public verifiable secret sharing as introduced in
+// "A Simple Publicly Verifiable Secret Sharing Scheme and its Application to
+// Electronic Voting" by Berry Schoenmakers. In comparison to regular verifiable
+// secret sharing schemes, PVSS enables any third party to verify shares
+// distributed by a dealer using zero-knowledge proofs.
+package pvss
+
+import (
+	"errors"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/proof"
+	"github.com/dedis/crypto/random"
+	"github.com/dedis/crypto/share"
+)
+
+// Some error definitions
+var errorTooFewShares = errors.New("not enough shares to recover secret")
+var errorDifferentLengths = errors.New("inputs of different lengths")
+var errorEncVerification = errors.New("verification of encrypted share failed")
+var errorDecVerification = errors.New("verification of decrypted share failed")
+
+// PubVerShare is a public verifiable share.
+type PubVerShare struct {
+	S share.PubShare  // Share
+	P proof.DLEQProof // Proof
+}
+
+// EncShares creates encrypted PVSS shares using the public keys X and
+// provides a NIZK encryption consistency proof for each share.
+func EncShares(suite abstract.Suite, H abstract.Point, X []abstract.Point, secret abstract.Scalar, t int) ([]*PubVerShare, *share.PubPoly, error) {
```

Comment:
Good thing I haven't touched that part for a while because now I don't understand what this functions does anymore and the comments are not helping. It'd be nice to have a descriptions of:
 + what is H
 + what is X
 + what is secret
 + what are the 2 values as output